File: baekjoon/1260번_DFS와BFS.cpp
#include <bits/stdc++.h>
#define IN_OPT std::ios_base::sync_with_stdio(false),std::cin.tie(nullptr)
using namespace std;

bool visited[1001];
vector<int> edges[1001];

void dfs(int s) {
    if (visited[s]) return;
    visited[s] = true;
    cout << s << " ";
    for (int i = 0; i < edges[s].size(); i++) {
        dfs(edges[s][i]);
    }
}

void bfs(int s) {
    queue<int> q;
    q.push(s);
    while (!q.empty()) {
        int p = q.front();
        q.pop();
        if (visited[p]) continue;
        visited[p] = true;
        cout << p << " ";
        for (int i = 0; i < edges[p].size(); i++) {
            q.push(edges[p][i]);
        }
    }
}

int main() {
    IN_OPT;
    int v, e, s;
    cin >> v >> e >> s;
    for (int i = 0; i < e; i++) {
        int v1, v2; cin >> v1 >> v2;
        edges[v1].push_back(v2);
        edges[v2].push_back(v1);
    }
    for (int i = 0; i < v; i++) {
        sort(edges[i].begin(), edges[i].end());
    }
    
    dfs(s); cout << "\n";
    fill_n(visited, 1001, false);
    bfs(s);
}

File: baekjoon/1260번_DFS와BFS.kt
import java.util.*

val visited = Array(1001) { false }
val edges = Array<MutableList<Int>>(1001) { mutableListOf() }

fun dfs(v: Int) {
    if (visited[v]) return
    visited[v] = true

    print("$v ")
    edges[v].forEach { dfs(it) }
}

fun bfs(v: Int) {
    val q: Queue<Int> = LinkedList()
    q.add(v)

    while(!q.isEmpty()) {
        val p = q.poll()
        if (visited[p]) continue
        visited[p] = true

        print("$p ")
        edges[p].forEach { q.add(it) }
    }
}

fun main() = with(Scanner(System.`in`)) {
    val n = nextInt()
    val m = nextInt()
    val v = nextInt()

    for(i in 0 until m) {
        val v1 = nextInt()
        val v2 = nextInt()
        edges[v1].add(v2)
        edges[v2].add(v1)
    }

    edges.forEach { it.sort() }
    dfs(v)
    println()
    visited.fill(false)
    bfs(v)
}